array-destructuring/exercise-3/exercise.js

```diff
@@ -2,7 +2,19 @@ let order = [
   { itemName: "Hot cakes", quantity: 1, unitPrice: 2.29 },
   { itemName: "Apple Pie", quantity: 2, unitPrice: 1.39 },
   { itemName: "Egg McMuffin", quantity: 1, unitPrice: 2.8 },
-  { itemName: "Sausage McMuffin", quantity: 1, unitPrice: 3.0 },
+  { itemName: "Sausage Muff", quantity: 1, unitPrice: 3.0 },
   { itemName: "Hot Coffee", quantity: 2, unitPrice: 1.0 },
   { itemName: "Hash Brown", quantity: 4, unitPrice: 0.4 },
 ];
+
+function printBill(list) {
+  let total = 0;
+  console.log("QTY", "\t", "ITEM", "\t\t", "TOTAL");
+  list.map(({ quantity, itemName, unitPrice }) => {
+    console.log(quantity, "\t", itemName, "\t", unitPrice * quantity);
+    total += quantity * unitPrice;
+  });
+  console.log("Total:", total);
+}
+
+printBill(order);
```

Comment:
Good job!

Also, an improvement you could make to this solution would be to use Array.reduce() to create a running aggregate instead of defining totalPrice outside of the loop.

Here is a link that explains using reduce to calculate a sum.

https://linuxhint.com/call-reduce-on-an-array-of-objects-to-sum-their-properties/

Reply:
that's a really useful method. I wish I knew the importance of code review and learn from it long before. Thank you very much for your brilliant and constructive feedback. Sorry for replying after a month. I never knew learning from code review would be this effective. 🙏